src/reducers/common.js

```diff
@@ -5,5 +5,9 @@ const updateActivityList = (flag) => {
   return (list) => list.pop();
 };
 
+const updateDatasets = (payload) => {
```

Comment:
Since you are only pushing a dataset, maybe go with `pushDatasetToList` or something similar.